### PR TITLE
Setup of basic React + MUI + Jest + eslint project

### PR DESCRIPTION
**Pull request by JarydMeek**
This is basically a "checkpoint" of sorts that contains everything necessary for a basic, empty React project.

## What's Included:
- React
  - Typescript, with strict typing enabled, please refrain from using "any" wherever possible
- MUI
- ESLint
  - Setup some pretty aggressive rules, but it will make our code significantly better over time
- Jest
  - One basic test to make sure everything works.

### CODE REVIEW

File: frontend/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        format: ['PascalCase'],
+      },
+    ],
+  },
+  "settings": {
+    "react": {
+        "version": "detect"
+        }
+    }
+};
```
Comment (computerdane) [inline frontend/.eslintrc.js:65]: why is this so complicated, why not just "extends": "airbnb-typescript"?

File: frontend/.gitignore [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (computerdane) [inline frontend/.gitignore:23]: We should probably just copy [this](https://github.com/github/gitignore/blob/main/Node.gitignore)

**Review by computerdane — APPROVED** [review]

File: frontend/.gitignore (hunk 2 above)
Comment (JarydMeek) [reply frontend/.gitignore:23]: Probably, that's just the default from `create-react-app` so I just left it, will include on next pr.

File: frontend/.eslintrc.js (hunk 1 above)
Comment (JarydMeek) [reply frontend/.eslintrc.js:65]: There are lots of things I prefer to override on `airbnb-typescript` to create a cleaner work environment, if you have any concerns with any specific rules I implemented, lmk and we can discuss

**Pull request merged**